x/auction/keeper/auctions.go

```diff
@@ -0,0 +1,337 @@
+	}
+
+	// Move Coins
+	increment := bid.Sub(a.Bid)
+	bidAmtToReturn := a.Bid
+	if bidder.Equals(a.Bidder) { // catch edge case of someone updating their bid with a low balance
+		bidAmtToReturn = sdk.NewInt64Coin(a.Bid.Denom, 0)
+	}
+	err := k.supplyKeeper.SendCoinsFromAccountToModule(ctx, bidder, types.ModuleName, sdk.NewCoins(bidAmtToReturn.Add(increment)))
+	if err != nil {
+		return a, err
+	}
+	err = k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, a.Bidder, sdk.NewCoins(bidAmtToReturn))
+	if err != nil {
+		return a, err
+	}
+	err = k.supplyKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, a.Initiator, sdk.NewCoins(increment)) // increase in bid size is burned
+	if err != nil {
+		return a, err
+	}
+	err = k.supplyKeeper.BurnCoins(ctx, a.Initiator, sdk.NewCoins(increment))
+	if err != nil {
+		return a, err
+	}
+
+	// Update Auction
+	a.Bidder = bidder
+	a.Bid = bid
+	// increment timeout
+	a.EndTime = earliestTime(ctx.BlockTime().Add(types.DefaultBidDuration), a.MaxEndTime)
+
+	return a, nil
+}
+func (k Keeper) PlaceBidForwardReverse(ctx sdk.Context, a types.ForwardReverseAuction, bidder sdk.AccAddress, bid sdk.Coin, lot sdk.Coin) (types.ForwardReverseAuction, sdk.Error) {
+	// Validate New Bid // TODO min bid increments, make validation code less confusing
+	if !a.Bid.IsEqual(a.MaxBid) {
+		// Auction is in forward phase, a bid here can put the auction into forward or reverse phases
+		if !a.Bid.IsLT(bid) {
+			return a, sdk.ErrInternal("auction in forward phase, new bid not higher than last bid")
+		}
+		if a.MaxBid.IsLT(bid) {
+			return a, sdk.ErrInternal("bid higher than max bid")
+		}
+		if lot.IsNegative() || a.Lot.IsLT(lot) {
+			return a, sdk.ErrInternal("lot out of bounds")
+		}
+		if lot.IsLT(a.Lot) && !bid.IsEqual(a.MaxBid) {
+			return a, sdk.ErrInternal("auction cannot enter reverse phase without bidding max bid")
+		}
+	} else {
+		// Auction is in reverse phase, it can never leave reverse phase
+		if !bid.IsEqual(a.MaxBid) {
+			return a, sdk.ErrInternal("") // not necessary
+		}
+		if lot.IsNegative() {
+			return a, sdk.ErrInternal("can't bid negative amount")
+		}
+		if !lot.IsLT(a.Lot) {
+			return a, sdk.ErrInternal("auction in reverse phase, new bid not less than previous amount")
+		}
+	}
+
+	// Move Coins
+	bidIncrement := bid.Sub(a.Bid)
+	bidAmtToReturn := a.Bid
+	lotDecrement := a.Lot.Sub(lot)
+	if bidder.Equals(a.Bidder) { // catch edge case of someone updating their bid with a low balance
+		bidAmtToReturn = sdk.NewInt64Coin(a.Bid.Denom, 0)
+	}
+	err := k.supplyKeeper.SendCoinsFromAccountToModule(ctx, bidder, types.ModuleName, sdk.NewCoins(bidAmtToReturn.Add(bidIncrement)))
+	if err != nil {
+		return a, err
+	}
+	err = k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, a.Bidder, sdk.NewCoins(bidAmtToReturn))
+	if err != nil {
+		return a, err
+	}
+	err = k.supplyKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, a.Initiator, sdk.NewCoins(bidIncrement))
```

Comment:
`bidIncrement` is always 0 in reverse phase, resulting in a transfer of 0 coins. Could place inside a reverse vs. forward phase conditional to prevent this transfer (see below).